torch的版本有限制吗？训练时报错！求助解答

之前那个diffusers的问题没有了，出现了新的问题。求助解答
torch的版本有限制吗？我用的版本是torch==2.1.2+cu121
现在的报错信息：
TypeError: cannot pickle 'torch._C._distributed_c10d.ProcessGroup' object
[2024-03-06 10:13:06,886] torch.distributed.elastic.multiprocessing.api: [ERROR] failed (exitcode: 1) local_rank: 0 (pid: 52488) of binary: /root/miniconda3/bin/python

我怀疑是torch版本的问题

--- a/train.py
+++ b/train.py
@@ -9,7 +9,6 @@ A minimal training script for DiT using PyTorch DDP.
 """
 import argparse
 import os
-from copy import deepcopy
 from functools import partial
 
 import torch
@@ -119,7 +118,8 @@ def main(args):
     # Step 4: Create DiT and EMA
     model = DiT_models[args.model](**model_kwargs).to(get_current_device())
     patch_size = model.patch_size
-    ema = deepcopy(model)
+    ema = DiT_models[args.model](**model_kwargs).to(get_current_device())
+    update_ema(ema, model, decay=0)
     requires_grad(ema, False)
 
     model.train()  # important! This enables embedding dropout for classifier-free guidance

